File: kotlin/src/main/kotlin/com/joshy/hywrapper/data/skyblock/GardenResource.kt
package com.joshy.hywrapper.data.skyblock

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class GardenResource(val cleanName: String) {
    @SerialName("WHEAT")
    WHEAT("Wheat"),

    @SerialName("CARROT_ITEM")
    CARROT("Carrot"),

    @SerialName("POTATO_ITEM")
    POTATO("Potato"),

    @SerialName("PUMPKIN")
    PUMPKIN("Pumpkin"),

    @SerialName("MELON")
    MELON("Melon"),

    @SerialName("INK_SACK:3")
    COCOA_BEANS("Cocoa Beans"),

    @SerialName("DOUBLE_PLANT")
    SUNFLOWER("Sunflower"),

    @SerialName("SUGAR_CANE")
    SUGAR_CANE("Sugar Cane"),

    @SerialName("MUSHROOM_COLLECTION")
    MUSHROOM("Mushroom"),

    @SerialName("CACTUS")
    CACTUS("Cactus"),

    @SerialName("NETHER_STALK")
    NETHER_WART("Nether Wart"),

    @SerialName("MOONFLOWER")
    MOONFLOWER("Moonflower"),

    @SerialName("WILD_ROSE")
    WILD_ROSE("Wild Rose")
}

File: python/src/hywrapper/data/skyblock/garden_resource.py
from __future__ import annotations

from enum import Enum
from typing import Any, Optional


class GardenResource(str, Enum):
    WHEAT = "WHEAT"
    CARROT = "CARROT_ITEM"
    POTATO = "POTATO_ITEM"
    PUMPKIN = "PUMPKIN"
    MELON = "MELON"
    COCOA_BEANS = "INK_SACK:3"
    SUNFLOWER = "DOUBLE_PLANT"
    SUGAR_CANE = "SUGAR_CANE"
    MUSHROOM = "MUSHROOM_COLLECTION"
    CACTUS = "CACTUS"
    NETHER_WART = "NETHER_STALK"
    MOONFLOWER = "MOONFLOWER"
    WILD_ROSE = "WILD_ROSE"

    @property
    def clean_name(self) -> str:
        return _CLEAN_NAMES[self]

    @classmethod
    def from_id(cls, id: Any) -> Optional[GardenResource]:
        try:
            return cls(id)
        except ValueError:
            return None


_CLEAN_NAMES = {
    GardenResource.WHEAT: "Wheat",
    GardenResource.CARROT: "Carrot",
    GardenResource.POTATO: "Potato",
    GardenResource.PUMPKIN: "Pumpkin",
    GardenResource.MELON: "Melon",
    GardenResource.COCOA_BEANS: "Cocoa Beans",
    GardenResource.SUNFLOWER: "Sunflower",
    GardenResource.SUGAR_CANE: "Sugar Cane",
    GardenResource.MUSHROOM: "Mushroom",
    GardenResource.CACTUS: "Cactus",
    GardenResource.NETHER_WART: "Nether Wart",
    GardenResource.MOONFLOWER: "Moonflower",
    GardenResource.WILD_ROSE: "Wild Rose",
}
